routes/project.js

```diff
@@ -250,18 +251,28 @@ router.get('/:branch/*.xslx', function(req, res, next) {
 	res.send('I am xslx for ' + req.params[0]);
 });
 
-router.get('/:branch/history', function(req, res, next) {
+router.get('/:branch/history/:limit?', function(req, res, next) {
 	var branchR = req.params.branch || req.params.commit;
 	var branch = branchR || 'master';
 	branch = branch.replace(/^~/, ''); //remove leading ~ when we match with branch with command
 	
 	var area = req.params.area;
-	var project = req.params.project;
+	var project = req.params.project;	
+	var limit = parseInt(req.params.limit || 20);
 
 	var title = area + '/' + project + (branchR ? '/' + branchR : '')
-	history(req, branch, function(err, commits) {
+	history(req, branch, limit+1, function(err, commits) {
```

Comment:
Ask for one more item than needed.  Pop it and use the URL as the pointer for the next history page